xgcm/test/test_grid.py

```diff
@@ -663,3 +659,26 @@ def test_multi_axis_input(all_datasets, func, periodic, boundary):
             serial = getattr(grid, func)(serial, axis, boundary=boundary_axis)
         full = getattr(grid, func)(ds[varname], axes, boundary=boundary)
         xr.testing.assert_allclose(serial, full)
+
+
+# @pytest.mark.parametrize("funcname", ["diff", "interp", "min", "max", "integrate", "average", "cumint"])
+# integrate, average and cumsum call xarray sum method directly and
+# not _wrap_and_replace_coords
+@pytest.mark.parametrize("funcname", ["diff", "interp", "min", "max"])
+def test_keep_coords(funcname):
+    # ds, periodic, expected = all_datasets
+    ds, coords, metrics = datasets_grid_metric("C")
+    ds = ds.assign_coords(yt_bis=ds["yt"], xt_bis=ds["xt"])
+    grid = Grid(ds, coords=coords, metrics=metrics)
+    func = getattr(grid, funcname)
+    # print(ds)
+    for axis_name in grid.axes.keys():
+        coords = list(func(ds.tracer, axis_name).coords)
+        assert "xt_bis" not in coords
+        assert "yt_bis" not in coords
+        #
```

Comment:
I like this way of testing that the dimension coordinates are in fact dropped, but I would suggest to borrow some logic from the actual implementation here and be even more explicit. E.g. `test_coords = [c for c in ds.coords if set(c.dims)==set(result.dims)]` where `results=func(ds.tracer, axis_name, keep_coords=True)`. The assert that all the test_coords are in results. 

I also think we should explicitly test for them being dropped if `keep_coords=False`, this will help with bookkeeping between the different methods.

Reply:
This comments raises a doubt in mind that needs clarification.
My understanding of the functionality we are trying to implement is that all coordinates of the *DataArray* that is passed to xgcm functions such as `grid` and that are compatible with the output DataArray are transfered.
The test you indicate (`test_coords = [c for c in ds.coords if set(c.dims)==set(result.dims)]`) suggests you'd like all compatible coordinates from the original *Dataset* should be transfered. Is that what you think should happen?
If yes, I don't see how this can be possible as to my understanding all coordinates originally not compatible with the input DataArray are dropped immediately upon selection of the DataArray from the Dataset.

Reply:
You are absolutely right. This was sloppy notation (I sketched the function without actually running anything). I think I meant to use `ds.tracer` instead of `ds`. Does that make more sense?
Apologies for the confusion.

Reply:
This looks great now. Very readable implementation.